### ISSUE TITLE
Cannot parse computed columns in the presence of identity column

### ISSUE THREAD

**Issue by asm0dey** [jOOQ/jOOQ#13689]
When working with following table:
```sql
create table author
(
    id          INTEGER primary key autoincrement,
    fb2id       TEXT,
    first_name  TEXT,
    middle_name TEXT,
    last_name   TEXT,
    nickname    TEXT,
    added       TEXT not null,
    gen_fnnn        generated always as (coalesce(first_name, '')),
    gen_lnnn        generated always as (coalesce(last_name, '')),
    gen_mnnn        generated always as (coalesce(middle_name, '')),
    gen_nnnn        generated always as (coalesce(nickname, '')),
    UNIQUE (gen_fnnn, gen_lnnn, gen_mnnn, gen_nnnn)
)
```
I encountered following error:
```
> Task :generateJooq
15:06:19 SEVERE An exception occurred while parsing a DDL script: Token ')' expected: [10:31] ...     TEXT not null,
    gen_fnnn        generated [*]always as (coalesce(first_name, '')),
    gen_lnnn        generated always as (c.... Please report this error to https://github.com/jOOQ/jOOQ/issues/new
org.jooq.impl.ParserException: Token ')' expected: [10:31] ...     TEXT not null,
    gen_fnnn        generated [*]always as (coalesce(first_name, '')),
    gen_lnnn        generated always as (c...
        at org.jooq.impl.DefaultParseContext.expected(ParserImpl.java:14001)
        at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:13257)
        at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:13252)
        at org.jooq.impl.DefaultParseContext.parseCreateTable(ParserImpl.java:4366)
        at org.jooq.impl.DefaultParseContext.parseCreate(ParserImpl.java:2795)
        at org.jooq.impl.DefaultParseContext.parseQuery(ParserImpl.java:1082)
        at org.jooq.impl.DefaultParseContext.lambda$parse$0(ParserImpl.java:874)
        at org.jooq.impl.DefaultParseContext.wrap(ParserImpl.java:14228)
        at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:854)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:732)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:727)
        at org.jooq.impl.InterpreterMetaProvider.loadSource(InterpreterMetaProvider.java:106)
        at org.jooq.impl.InterpreterMetaProvider.provide(InterpreterMetaProvider.java:92)
        at org.jooq.impl.SourceMetaProvider.provide(SourceMetaProvider.java:80)
        at org.jooq.impl.DefaultDSLContext.meta(DefaultDSLContext.java:460)
        at org.jooq.impl.MetaImpl$MetaSchema.lambda$initUksSQLite$5(MetaImpl.java:517)
        at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
        at org.jooq.impl.MetaImpl$MetaSchema.initUksSQLite(MetaImpl.java:504)
        at org.jooq.impl.MetaImpl$MetaSchema.getUks(MetaImpl.java:450)
        at org.jooq.impl.MetaImpl$MetaSchema.lambda$getTables$2(MetaImpl.java:434)
        at org.jooq.impl.Tools.map(Tools.java:2149)
        at org.jooq.impl.MetaImpl$MetaSchema.getTables(MetaImpl.java:406)
        at org.jooq.impl.Snapshot$SnapshotSchema.<init>(Snapshot.java:125)
        at org.jooq.impl.Snapshot$SnapshotCatalog.lambda$new$0(Snapshot.java:100)
        at org.jooq.impl.Tools.map(Tools.java:2149)
        at org.jooq.impl.Snapshot$SnapshotCatalog.<init>(Snapshot.java:100)
        at org.jooq.impl.Snapshot.lambda$getCatalogs0$0(Snapshot.java:92)
        at org.jooq.impl.Tools.map(Tools.java:2149)
        at org.jooq.impl.Snapshot.getCatalogs0(Snapshot.java:92)
        at org.jooq.impl.AbstractMeta.initCatalogs(AbstractMeta.java:124)
        at org.jooq.impl.AbstractMeta.getCatalogs(AbstractMeta.java:116)
        at org.jooq.impl.Snapshot.<init>(Snapshot.java:80)
        at org.jooq.impl.AbstractMeta.snapshot(AbstractMeta.java:537)
        at org.jooq.meta.sqlite.SQLiteDatabase.snapshot(SQLiteDatabase.java:449)
        at org.jooq.meta.sqlite.SQLiteDatabase.loadPrimaryKeys(SQLiteDatabase.java:220)
        at org.jooq.meta.AbstractDatabase.lambda$getRelations0$24(AbstractDatabase.java:2756)
        at org.jooq.meta.AbstractDatabase.onError(AbstractDatabase.java:3361)
        at org.jooq.meta.AbstractDatabase.getRelations0(AbstractDatabase.java:2756)
        at org.jooq.meta.AbstractDatabase.lambda$getRelations$17(AbstractDatabase.java:2438)
        at org.jooq.meta.AbstractDatabase.onError(AbstractDatabase.java:3361)
        at org.jooq.meta.AbstractDatabase.getRelations(AbstractDatabase.java:2438)
        at org.jooq.meta.DefaultColumnDefinition.getPrimaryKey(DefaultColumnDefinition.java:143)
        at org.jooq.meta.AbstractTableDefinition.getPrimaryKey(AbstractTableDefinition.java:96)
        at org.jooq.codegen.JavaGenerator.generateTable(JavaGenerator.java:5348)
        at org.jooq.codegen.JavaGenerator.generateTables(JavaGenerator.java:5329)
        at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:613)
        at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:568)
        at org.jooq.codegen.JavaGenerator.generate0(JavaGenerator.java:483)
        at org.jooq.codegen.AbstractGenerator.generate(AbstractGenerator.java:178)
        at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:202)
        at org.jooq.codegen.GenerationTool.run0(GenerationTool.java:904)
        at org.jooq.codegen.GenerationTool.run(GenerationTool.java:239)
        at org.jooq.codegen.GenerationTool.generate(GenerationTool.java:234)
        at org.jooq.codegen.GenerationTool.main(GenerationTool.java:206)
```

DB: sqlite
driver: org.xerial:sqlite-jdbc:3.36.0.3

**Comment by lukaseder** [comment]
Thanks a lot for your report. I can reproduce it. I think it's related to the `autoincrement` column, which of course shouldn't have such an effect.

Will investigate right away.

**Comment by lukaseder** [comment]
Note that the exact syntax supported by SQLite in the documentation is a bit weird. It claims that you can use `GENERATED ALWAYS` (which is a SQL standard), but you can really put anything you want:

```sql
create table t (i int, j geeeenerattteeeeed as (i));
pragma table_xinfo('t');
```

Producing

```
|cid|name|type              |notnull|dflt_value|pk |hidden|
|---|----|------------------|-------|----------|---|------|
|0  |i   |int               |0      |          |0  |0     |
|1  |j   |geeeenerattteeeeed|0      |          |0  |2     |
```

The essential thing is that you put `AS`, which already works, if you also repeat the type. Omitting the type isn't supported yet.

**Comment by lukaseder** [comment]
I've created https://github.com/jOOQ/jOOQ/issues/13713 for the omitted data type issue. Will fix the more pressing issue that prevents computed columns in the presence of `autoincrement`. Note, if you reverse the order of columns (and make the computed data type explicit), I think you should be able parse the statement.

Also note, computed columns are a commercial only feature in jOOQ.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.1 (https://github.com/jOOQ/jOOQ/issues/13714), and 3.16.8 (https://github.com/jOOQ/jOOQ/issues/13715)
